### PR TITLE
Keep the dialog state after returning to the home page

### PR DESCRIPTION
**Pull request by ixiumu**
Fix [#188](https://github.com/recloudstream/cloudstream/issues/188)

### CODE REVIEW

**Comment by Indranil012** [comment]
@Blatzar merge when 😁

File: app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt [hunk 1]
```diff
@@ -210,7 +213,13 @@ class HomeFragment : Fragment() {
             recycle.adapter = SearchAdapter(item.list.toMutableList(), recycle) { callback ->
                 handleSearchClickCallback(this, callback)
                 if (callback.action == SEARCH_ACTION_LOAD || callback.action == SEARCH_ACTION_PLAY_FILE) {
-                    bottomSheetDialogBuilder.dismissSafe(this)
+//                    bottomSheetDialogBuilder.dismissSafe(this)
+                    handler.postDelayed({
```
Comment (Blatzar) [inline app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt:217]: why delayed and why `it.window?.setWindowAnimations(-1)` ?
`bottomSheetDialogBuilder.hide()` does the job just fine

File: app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt [hunk 2]
```diff
@@ -257,6 +266,17 @@ class HomeFragment : Fragment() {
...
+            return bottomSheetDialogBuilder
+        }
+
+        fun BottomSheetDialog.onResume(context: Context?) {
+            // resume dialog
+            this.window?.setWindowAnimations(-1)
+            this.show()
+            handler.postDelayed({
+                this.window?.setWindowAnimations(R.style.BottomSheetDialog)
+            }, 400)
```
Comment (Blatzar) [inline app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt:279]: Why delayed and with animations?
`bottomSheetDialog?.show()` works just fine

File: app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt [hunk 3]
```diff
@@ -1099,9 +1125,10 @@ class HomeFragment : Fragment() {
             ParentItemAdapter(mutableListOf(), { callback ->
                 homeHandleSearch(callback)
             }, { item ->
-                activity?.loadHomepageList(item, expandCallback = {
+                bottomSheetDialog = activity?.loadHomepageList(item, expandCallback = {
                     homeViewModel.expandAndReturn(it)
                 })
+                bottomSheetDialog?.setOnDismissListener { bottomSheetDialog = null }
```
Comment (Blatzar) [inline app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt:1131]: Good :pray:

File: app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt [hunk 4]
```diff
@@ -154,10 +157,10 @@ class HomeFragment : Fragment() {
             expand: HomeViewModel.ExpandableHomepageList,
             deleteCallback: (() -> Unit)? = null,
             expandCallback: (suspend (String) -> HomeViewModel.ExpandableHomepageList?)? = null
-        ) {
+        ): BottomSheetDialog? {
             val context = this
-            val bottomSheetDialogBuilder = BottomSheetDialog(context)
-            bottomSheetDialogBuilder.setContentView(R.layout.home_episodes_expanded)
+            var bottomSheetDialogBuilder: BottomSheetDialog? = BottomSheetDialog(context)
```
[the remark this answers is not in the document]
Comment (ixiumu) [reply app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt:162]: If not, the dialog will be displayed unnecessarily in the search tab

File: app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt (hunk 1 above)
Comment (ixiumu) [reply app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt:217]: After clicking, the dialog will be hidden, and the new page will not be displayed immediately

File: app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt [hunk 5]
```diff
@@ -257,6 +266,17 @@ class HomeFragment : Fragment() {
             //(recycle.adapter as SearchAdapter).notifyDataSetChanged()
 
             bottomSheetDialogBuilder.show()
+
+            return bottomSheetDialogBuilder
+        }
+
+        fun BottomSheetDialog.onResume(context: Context?) {
```
Comment (Blatzar) [inline app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt:273]: context is unused?

**Review by Blatzar — CHANGES_REQUESTED** [review]
I understand your animation stuff now, but there are still some changes to be made :pensive:

File: app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt (hunk 5 above)
Comment (ixiumu) [reply app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt:273]: yes, not used, you can delete it

File: app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (ixiumu) [reply app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt:162]: Sorry, they were needed at first, I made some changes later
Now this function can return null

I have been testing the lazy loading of the homepage category in the local branch, and there are still some bugs.

This pull, just simply copy these short codes to the master
You can clean up useless code

they work fine for me

[the remark this answers is not in the document]
Comment (Blatzar) [reply app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt:162]: Lazy loading?

**Comment by Indranil012** [comment]
Please fix this, I beg you 🫠

**Comment by Indranil012** [comment]
since @ixiumu is inactive from long time, @Blatzar could you please take over this pr? 🙂

**Comment by LagradOst** [comment]
fixed myself

**Pull request closed without merging**